src/udfs/ssd_object_detector.py

```diff
@@ -0,0 +1,375 @@
+import itertools
+import numpy as np
+import pandas as pd
+
+from typing import List
+from math import sqrt
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+from torchvision.transforms import Compose, transforms
+
+from src.models.catalog.frame_info import FrameInfo
+from src.models.catalog.properties import ColorSpace
+from src.models.inference.outcome import Outcome
+from src.udfs.pytorch_abstract_udf import PytorchAbstractUDF
+
+
+class SSDObjectDetector(PytorchAbstractUDF):
+    @property
+    def name(self) -> str:
+        return "ssd"
+
+    def __init__(self, threshold=0.5):
+        super().__init__()
+        self.threshold = threshold
+
+        # load ssd from pytorch hub api and default to cpu
+        self.model = torch.hub.load(
+            "NVIDIA/DeepLearningExamples:torchhub",
+            "nvidia_ssd",
+            pretrained=False)
+        model_state_dict = torch.hub.load_state_dict_from_url(
+            "https://api.ngc.nvidia.com/v2/models/nvidia/ssdpyt_fp32/versions/1/files/nvidia_ssdpyt_fp32_20190225.pt",
+            map_location=torch.device("cpu"))["model"]
+        self.model.load_state_dict(model_state_dict)
+        self.model.eval()
+
+    @property
+    def labels(self) -> List[str]:
+        return [
+            "__background__", "person", "bicycle", "car", "motorcycle",
+            "airplane", "bus",
+            "train", "truck", "boat", "traffic light", "fire hydrant",
+            "stop sign",
+            "parking meter", "bench", "bird", "cat", "dog", "horse", "sheep",
+            "cow",
+            "elephant", "bear", "zebra", "giraffe", "backpack", "umbrella",
+            "handbag", "tie", "suitcase", "frisbee", "skis", "snowboard",
+            "sports ball",
+            "kite", "baseball bat", "baseball glove", "skateboard",
+            "surfboard", "tennis racket",
+            "bottle", "wine glass", "cup", "fork", "knife", "spoon",
+            "bowl",
+            "banana", "apple", "sandwich", "orange", "broccoli", "carrot",
+            "hot dog", "pizza",
+            "donut", "cake", "chair", "couch", "potted plant", "bed",
+            "dining table",
+            "toilet", "tv", "laptop", "mouse", "remote", "keyboard",
+            "cell phone",
+            "microwave", "oven", "toaster", "sink", "refrigerator", "book",
+            "clock", "vase", "scissors", "teddy bear", "hair drier",
+            "toothbrush"
+        ]
+
+    @property
+    def input_format(self) -> FrameInfo:
+        return FrameInfo(-1, -1, 3, ColorSpace.RGB)
+
+    @property
+    def transforms(self) -> Compose:
+        return Compose([
+            transforms.Resize([300, 300]),
+            transforms.ToTensor(),
+        ])
+
+    def _get_predictions(self, frames: Tensor) -> List[Outcome]:
```

Comment:
@xzdandy  I think pd.Dataframe is more generic. We should remove the Outcome all together from the repository.

Reply:
So it should be a list of panda frames?

Reply:
def classify(self, frames: Tensor) -> pd.DataFrame:, 
panda dataframe is a table. Every row is oen result.

Reply:
I see. Basically I was looking [here](https://github.com/georgia-tech-db/eva-reuse/blob/df6df0251361a85e6b4cd2123db8770affa76f6e/src/udfs/fastrcnn_object_detector.py#L108), but I got confused because every branch has it own format 🙂 .

Reply:
Ah. I see. eva_reuse haven't merged the latest master branch.